src/tools/compiletest/src/read2.rs

```diff
@@ -0,0 +1,208 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// FIXME: This is a complete copy of `cargo/src/cargo/util/read2.rs`
```

Comment:
Another option (which Cargo originally did) is to just spawn threads (libstd actually used to do this too). This was only switched to `read2` as an optimization later, but compiletest can probably get away with threads and not worry to much about perf if you're looking to fix this.

Reply:
@alexcrichton Well copy-and-paste here is simpler than writing new code :p

Reply:
Makes sense!